docs/optlib.rst

```diff
@@ -569,6 +570,144 @@ Output::
 
 "X.y" is printed as a fully qualified tag when ``--extras=+q`` is given.
 
+.. NOT REVIEWED YET (--_scopesep)
+
+Customizing scope separators
+......................................................................
+Use ``--_scopesep-<LANG>=<child-kletter>,[<parent-kletter>]:<sep>``
+option for customizing.
+
+``child-kletter``
+
+	The kind letter for a tag of inner scope.
+
+	You can use `*` for specifying as wildcards that means
+	"any kinds" for a tag of inner scope.
+
+``parent-kletter``
+
+	The kind letter for a tag of outer scope.
+
+	You can use `*` for specifying as wildcards that means
+	"any kinds" for a tag of outer scope.
+
+	If you omit ``parent-kletter``, the separator used as
+	prefix for tags specified with ``child-kletter``.
+
+``sep``
+
+
+	Separator used for the combination.
+
+Specifying `*` as both ``child-kletter`` and ``parent-kletter`` sets
+``sep`` as the language default separator. It is used as fallback.
+
+Specifying `*` as ``child-kletter`` and omitting ``parent-kletter``
+sets ``sep`` as the language default prefix. It is used as fallback.
+
+NOTE: There is no ctags global default prefix.
+NOTE: ``_scopesep-<LANG>=...`` option affects only a parser that
+enables ``_autoFQTag``. A parser building full qualified tags
+manually ignores the option.
+
+Let's see an example.
+The input file is written in Tcl.
+Tcl parser uses the ``_autoFQTag`` feature internally though it is
+written in crafted C code.
```

Comment:
``{_autoFQTag}`` is a flag that can be specified from command line.
puppetManifest parser specifies it, as you may know.
We call a parser written in the command line interface provided by ctags an optlib parser.
The tcl parser is not optlib parser. It is written in C manually.
"manually" is a point because an optlib parser is converted to C program in ctags building process.

A program written in C manually is called "crafted" parser. If you find a better name, let me know.
Tcl parser uses code behind ``{_autoFQTag}}`` directly.